### PR TITLE
Add missing dependency @types/hoist-non-react-statics, closes #3837

### LINKED ISSUE #3837: Type error compiling with typescript
### Current Behavior

Typescript compilation fails with type errors in Formik library
```
<project>/node_modules/formik/dist/ErrorMessage.d.ts(1,23): error TS2688: Cannot find type definition file for 'hoist-non-react-statics'.
<project>/node_modules/formik/dist/FastField.d.ts(1,23): error TS2688: Cannot find type definition file for 'hoist-non-react-statics'.
<project>/node_modules/formik/dist/FieldArray.d.ts(1,23): error TS2688: Cannot find type definition file for 'hoist-non-react-statics'.       
```

The npm package `hoist-non-react-statics` hasn't been updated in years, so seems unlikely to be an upstream issue.

Typescript version 4.9.5.

### Expected behavior

Compilation should suceed without type errors.

### Reproducible example

I don't have an easily shared reproduction, but this is my package-lock info:

```
"node_modules/formik": {
			"version": "2.4.2",
			"resolved": "https://registry.npmjs.org/formik/-/formik-2.4.2.tgz",
			"integrity": "sha512-C6nx0hifW2uENP3M6HpPmnAE6HFWCcd8/sqBZEOHZY6lpHJ5qehsfAy43ktpFLEmkBmhiZDei726utcUB9leqg==",
			"funding": [
				{
					"type": "individual",
					"url": "https://opencollective.com/formik"
				}
			],
			"dependencies": {
				"deepmerge": "^2.1.1",
				"hoist-non-react-statics": "^3.3.0",
				"lodash": "^4.17.21",
				"lodash-es": "^4.17.21",
				"react-fast-compare": "^2.0.1",
				"tiny-warning": "^1.0.2",
				"tslib": "^2.0.0"
			},
			"peerDependencies": {
				"react": ">=16.8.0"
			}
		},

```

react-hoist-non-statics is satisifed with 3.3.2, as it has been for a while:

```
		"hoist-non-react-statics": {
			"version": "3.3.2",
			"resolved": "https://registry.npmjs.org/hoist-non-react-statics/-/hoist-non-react-statics-3.3.2.tgz",
			"integrity": "sha512-/gGivxi8JPKWNm/W0jSmzcMPpfpPLc3dY/6GxhX2hQ9iGj3aDfklV4ET7NjKpSinLpJ5vafa9iiGIEZg10SfBw==",
			"requires": {
				"react-is": "^16.7.0"
			}
		},
```

### Suggested solution(s)

Not sure if this requires manual typing or setting a typescript version dependency. 

### Additional context

This happened after upgrading from Formik 2.4.1 to 2.4.2.

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Formik           |  2.4.2
| React            |  16.14.0
| TypeScript       | 4.9.5
| Browser          |  N/A
| npm/Yarn         | 8.5.5
| Operating System | windows 11 21H2

### PR DESCRIPTION
**Pull request by patik**
This package is needed by Formik but it was not included in the dependency list.

Fixes #3837

### CODE REVIEW

**Review by probablyup — APPROVED** [review]
Thank you

**Comment by probablyup** [comment]
Could you add a changeset please?

**Review by probablyup — APPROVED** [review]
Thanks!

**Review by probablyup — CHANGES_REQUESTED** [review]
Actually sorry, I think this type needs to go in formik's specific package.json not the top one?

**Comment by patik** [comment]
> Actually sorry, I think this type needs to go in formik's specific package.json not the top one?

Do you mean `packages/formik/package.json`? This dependency is already listed in that package (it's been there for at least 4 years), but the issue seems to persist unless the dependency is in the root-level package.json.

**Comment by probablyup** [comment]
> > Actually sorry, I think this type needs to go in formik's specific package.json not the top one?
> 
> Do you mean `packages/formik/package.json`? This dependency is already listed in that package (it's been there for at least 4 years), but the issue seems to persist unless the dependency is in the root-level package.json.

You're seeing that a dependency in a workspace package isn't installing downstream in your client package because it's not present in the workspace root package.json? That seems unlikely.

**Comment by patik** [comment]
> It's currently in `devDependencies`, perhaps move it to `dependencies` in that file?

Indeed, that change resolves the issue. I made the change, and now if I install Formik in another project, the `@types/hoist-non-react-statics` package is also installed, and the TS error goes away.

File: package.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
     "@playwright/test": "^1.34.3",
     "@types/benchmark": "^2.1.2",
     "@types/jest": "^25.0.0",
+    "@types/hoist-non-react-statics": "^3.3.1",
```
Comment (probablyup) [inline package.json:16]: Thanks, this one can be removed now

Comment (patik) [reply package.json]: Yep, fixed.

**Review by probablyup — APPROVED** [review]
Awesome, thank you :)

**Comment by probablyup** [comment]
Pinged Jared to fix the required check settings, please hold

**Pull request merged**